ably/test_config.go

```diff
@@ -53,6 +59,22 @@ func getEnvWithDefault(name string, defaultValue string) string {
 	return defaultValue
 }
 
+func ablyPublisher() bool {
+	return getEnvWithDefault("ABLY_PUBLISHER", DefaultPublisher) == "true"
+}
+
+func ablyNumChannels() int {
+	value := getEnvWithDefault("ABLY_NUM_CHANNELS", DefaultNumChannels)
+
+	n, err := strconv.Atoi(value)
+
+	if err != nil {
+		panic("Expected an Integer for 'ABLY_NUM_CHANNELS' - got '" + value + "'")
```

Comment:
It's probably better to avoid calling panic inside code. I don't think it's a terrible problem here, but writing code that panics and is tested by recovering from panics should be avoided. It would be better to have the config return an error and have the main function call `log.Fatalf`. It might also be better to inject an environment function than to depend on it explicitly, as this will aid with parallel testing. https://github.com/uber-go/guide/blob/master/style.md#dont-panic